tests/test_hdf.py

```diff
@@ -71,36 +71,72 @@ def test_field_capture_pcap_bits():
         units=None,
     )
 
-    assert pcap_bits_frame_data.is_pcap_bits_or_samples
+    assert not pcap_bits_frame_data.has_scale_or_offset
     assert pcap_bits_frame_data.raw_mode_dataset_dtype is np.dtype("uint32")
 
-    some_other_frame_data = FieldCapture(
-        name="some_other_frame_data",
+    frame_data_without_scale_offset = FieldCapture(
+        name="frame_data_without_scale_offset",
         type=np.dtype("uint32"),
         capture="Value",
         scale=1.0,
         offset=0.0,
         units="",
     )
 
-    assert not some_other_frame_data.is_pcap_bits_or_samples
-    assert some_other_frame_data.raw_mode_dataset_dtype is np.dtype("float64")
+    assert not frame_data_without_scale_offset.has_scale_or_offset
+    assert frame_data_without_scale_offset.raw_mode_dataset_dtype is np.dtype("float64")
 
-    malformed_frame_data = FieldCapture(
-        name="malformed_frame_data",
+    with pytest.raises(
+        ValueError,
+        match=(
+            "If any of `scale=None`, `offset=0.0`, or "
+            "`units=` is set, all must be set"
+        ),
+    ):
+        _ = FieldCapture(
+            name="malformed_frame_data",
+            type=np.dtype("uint32"),
+            capture="Value",
+            scale=None,
+            offset=0.0,
+            units="",
+        )
+
+    frame_data_with_offset = FieldCapture(
+        name="frame_data_with_offset",
         type=np.dtype("uint32"),
         capture="Value",
-        scale=None,
+        scale=1.0,
+        offset=1.0,
+        units="",
+    )
+    frame_data_with_scale = FieldCapture(
+        name="frame_data_with_scale",
+        type=np.dtype("uint32"),
+        capture="Value",
+        scale=1.1,
         offset=0.0,
         units="",
     )
 
-    assert not some_other_frame_data.is_pcap_bits_or_samples
-    with pytest.raises(
-        ValueError,
-        match="If any of `scale`, `offset`, or `units` is set, all must be set",
-    ):
-        assert malformed_frame_data.raw_mode_dataset_dtype is np.dtype("float64")
+    assert frame_data_with_offset.has_scale_or_offset
+    assert frame_data_with_offset.raw_mode_dataset_dtype is np.dtype("float64")
+    assert frame_data_with_scale.has_scale_or_offset
+    assert frame_data_with_scale.raw_mode_dataset_dtype is np.dtype("float64")
+
+    frame_data_with_scale_and_offset = FieldCapture(
+        name="frame_data_with_scale_and_offset",
+        type=np.dtype("uint32"),
```

Comment:
Please can we make the tests with scale and offset have dtype float64, and the ones without scale and offset uint32